File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/StartsWithComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import com.ekino.oss.jcv.core.JsonValueComparator;

import org.skyscreamer.jsonassert.ValueMatcherException;

import static java.util.Objects.*;

public class StartsWithComparator implements JsonValueComparator<String> {

    private final String value;

    public StartsWithComparator(String value) {
        this.value = requireNonNull(value);
    }

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        if (actual != null && actual.startsWith(value)) {
            return true;
        }
        throw new ValueMatcherException("Value should start with '" + value + "'", expected, actual);
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/StartsWithComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException

class StartsWithComparator(private val value: String) : JsonValueComparator<String> {

    override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
        if (actual != null && actual.startsWith(value)) {
            return true
        }
        throw ValueMatcherException("Value should start with '$value'", expected, actual)
    }
}
